Plugins/BundleEditorPlugin/BundleEditor.cs

```diff
@@ -11,9 +12,286 @@
+            entry.LinkAsset(MovieChunkEntry);
+
+            ChunkAssetEntry SubtitleChunkEntry = App.AssetManager.GetChunkEntry(movieobject.SubtitleChunkGuid);
+            if (SubtitleChunkEntry != null)
+            {
+                SubtitleChunkEntry.AddedBundles.Remove(App.AssetManager.GetBundleId(bentry));
+                entry.LinkAsset(SubtitleChunkEntry);
+            }
+        }
+    }
+
+    public class RemoveSoundWaveExtension : RemoveFromBundleExtension
+    {
+        public override string AssetType => "SoundWaveAsset";
+
+        public override void RemoveFromBundle(EbxAssetEntry entry, BundleEntry bentry)
+        {
+            base.RemoveFromBundle(entry, bentry);
+
+            EbxAsset soundasset = App.AssetManager.GetEbx(entry);
+            dynamic soundobject = soundasset.RootObject;
+
+            foreach (var soundChunk in soundobject.Chunks)
+            {
+                ChunkAssetEntry ChunkEntry = App.AssetManager.GetChunkEntry(soundChunk.ChunkId);
+                ChunkEntry.AddedBundles.Remove(App.AssetManager.GetBundleId(bentry));
+                entry.LinkAsset(ChunkEntry);
+            }
+        }
+    }
+
+    public class RemoveFromBundleExtension
+    {
+        public virtual string AssetType => null;
+        public virtual void RemoveFromBundle(EbxAssetEntry entry, BundleEntry bentry)
+        {
+            entry.AddedBundles.Remove(App.AssetManager.GetBundleId(bentry));
+        }
+    }
+    #endregion
+
+    #region AddToBundleExtension
+    public class MeshExtension : AddToBundleExtension
+    {
+        public override string AssetType => "MeshAsset";
+        public override void AddToBundle(EbxAssetEntry entry, BundleEntry bentry)
+        {
+            base.AddToBundle(entry, bentry);
+
+            EbxAsset asset = App.AssetManager.GetEbx(entry);
+            dynamic meshAsset = asset.RootObject;
+
+            //Add res to BUNDLES AND LINK
+            ResAssetEntry resEntry = App.AssetManager.GetResEntry(meshAsset.MeshSetResource);
+            resEntry.AddToBundle(App.AssetManager.GetBundleId(bentry));
+            
+            entry.LinkAsset(resEntry);
+
+            MeshSet meshSetRes = App.AssetManager.GetResAs<MeshSet>(resEntry);
+            
+            //Double check if there are any LODs the mesh, if there are, bundle and link them
+            // J-Lyt | If chunk is in SuperBundle, do not add to bundle.
+            if (meshSetRes.Lods.Count > 0)
+            {
+                foreach (MeshSetLod lod in meshSetRes.Lods)
+                {
+                    ChunkAssetEntry chunkEntry = App.AssetManager.GetChunkEntry(lod.ChunkId);
+                    if (lod.ChunkId != Guid.Empty && chunkEntry.SuperBundles.Count == 0)
+                    {
+                        chunkEntry.AddToBundle(App.AssetManager.GetBundleId(bentry));
+                        resEntry.LinkAsset(chunkEntry);
+                    }
+                }
+            }            
```

Comment:
Would it be better to add the empty Guid check first, before doing the retrieval?

```suggestion
            if (meshSetRes.Lods.Count > 0)
            {
                foreach (MeshSetLod lod in meshSetRes.Lods)
                {
                    if (lod.ChunkId != Guid.Empty)
                    {
                        ChunkAssetEntry chunkEntry = App.AssetManager.GetChunkEntry(lod.ChunkId);
                        if (chunkEntry != null && chunkEntry.SuperBundles.Count == 0)
                        {
                            chunkEntry.AddToBundle(App.AssetManager.GetBundleId(bentry));
                            resEntry.LinkAsset(chunkEntry);
                        }
                    }
                }
            }      
```